include/deal.II/matrix_free/fe_point_evaluation.h

```diff
@@ -2275,11 +2322,39 @@ FEPointEvaluation<n_components_, dim, spacedim, Number>::integrate(
       return;
     }
 
+  if (do_JxW)
+    Assert(
+      JxW_ptr != nullptr,
+      ExcMessage(
+        "JxW pointer is not set! If you do not want to integrate() use test_and_sum()"));
```

Comment:
```suggestion
    Assert(
    !do_JxW ||
      JxW_ptr != nullptr,
      ExcMessage(
        "JxW pointer is not set! If you do not want to integrate() use test_and_sum()"));
```

Reply:
We could also put it into the `integrate()` function since we have `do_integrate()` now? WDYT?

Reply:
Sounds good!